### PR TITLE
feat(v-list-tile): add hover prop

### LINKED ISSUE #4330: [Feature Request] Add "hover" prop to lists
### Problem to solve
Currently you have to add an empty click listener to lists in order for a hover-effect to be applied. This is used a lot in the docs as `@click=""`. This is confusing for people reading the code.

### Proposed solution
Add a `hover` prop.

### PR DESCRIPTION
**Pull request by zaken**
## Description
Add the `hover` prop to `v-list-tile` to enable link-like hover effect.

## Motivation and Context
Closes #4330 

## How Has This Been Tested?
Visually

## Markup:

<details>

```vue
<template>
  <v-app>
    <boilerplate>
      <h2 class="my-2">@click=""</h2>
      <v-list>
        <v-list-tile v-for="item in items" :key="item.title" @click="">
          <v-list-tile-content>
            <v-list-tile-title v-text="item.title"></v-list-tile-title>
          </v-list-tile-content>
        </v-list-tile>
      </v-list>
      <h2 class="my-2">hover</h2>
      <v-list>
        <v-list-tile v-for="item in items" :key="item.title" hover>
          <v-list-tile-content>
            <v-list-tile-title v-text="item.title"></v-list-tile-title>
          </v-list-tile-content>
        </v-list-tile>
      </v-list>
    </boilerplate>
  </v-app>
</template>

<script>
  export default {
    data () {
      return {
        items: [
          { title: 'Jason Oner' },
          { title: 'Travis Howard' },
          { title: 'Ali Connors' },
          { title: 'Cindy Baker' }
        ]
      }
    }
  }
</script>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

**Comment by jacekkarczmarczyk** [comment]
Please add an unit test

File: src/stylus/components/_lists.styl [hunk 1]
```diff
@@ -62,7 +63,8 @@ theme(v-list, "v-list")
     transition: $primary-transition
     user-select: none
 
-    &--link
+    &--link,
+    &--hover
```
Comment (jacekkarczmarczyk) [inline src/stylus/components/_lists.styl]: I'm not sure about this, it probably should just change the background but cursor shouldn't change as there is no any action

Comment (zaken) [reply src/stylus/components/_lists.styl]: Yeah I wasn't sure about this either. If there's any action `@click` would add the `cursor: pointer` anyway. Reverting.

**Comment by johnleider** [comment]
I'm thinking as to whether `v-hover` can satisfy this use-case. (v1.2 feature)

**Comment by johnleider** [comment]
Closing as this can be handled with `v-hover`.

**Comment by johnleider** [comment]
You will have to be patient. I don't particularly remember being messaged but their are plenty of developers helping in the help channel.

**Comment by happilymarrieddad** [comment]
@johnleider ok sorry man I didn't mean to be rude. Thanks so much!

**Pull request closed without merging**